xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/XWiki/DeletedDocuments.xml

```diff
@@ -349,7 +390,35 @@ document.observe('xwiki:dom:loaded', function() {
   if ($('documentsTrash')) {
     $('documentsTrash').up('.hidden').removeClassName('hidden');
   }
-});</code>
+});
+
+// we need to wait the complete display of the livetable to get the number of rows
+document.observe('xwiki:livetable:documentsTrash:displayComplete', function() {
+  if ($('documentsTrash')) {
+    var actionAllContainer = $('documentsTrash').
+        getElementsBySelector('td.xwiki-livetable-display-header-filter:last-child')[0];
+
+    if (trashDocumentsLivetable.totalRows > 0) {
+      var trashAllClass = 'btn btn-danger';
+    } else {
+      var trashAllClass = 'btn btn-danger disabled';
+    }
+
+    // avoid to get multiple times the button
+    if ($('trashAllDeleted')) {
+      $('trashAllDeleted').writeAttribute('class', trashAllClass);
+    } else {
+      var trashAll = new Element('a', {
+          'id': 'trashAllDeleted',
+          'class': trashAllClass,
+          'href':'#'
+          }).update('$services.localization.render('index.deletedDocuments.deleteAll')');
+      actionAllContainer.insert(trashAll);
+      trashAll.observe('click', XWiki.index.trash.documents.confirmDeleteAll.bindAsEventListener(trashAll));
+    }
```

Comment:
I find the above code a bit overly complicated. I would do this instead:

* get the button
* if the button doesn't exist then create it (enabled)
* update the button state (enabled / disabled) with:

        trashAll.toggleClassName('disabled', trashDocumentsLivetable.totalRows > 0);

Reply:
I'll refactor.